app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -25,6 +25,8 @@ module CollectionBehavior
         index.as(*index_collection_type_gid_as)
       end
 
+      after_find { |col| load_collection_type_instance(col) } # lazy migrate older collections to have collection type gid
```

Comment:
Can we avoid callbacks?  Callbacks like this break the single responsibility principal and  aren't going to work with Valkyrie (so we'd just remove it?).